src/main/java/com/desiato/puresynth/configurations/AuthTokenFilter.java

```diff
@@ -0,0 +1,66 @@
+package com.desiato.puresynth.configurations;
+
+import com.desiato.puresynth.exceptions.InvalidTokenException;
+import com.desiato.puresynth.models.CustomUserDetails;
+import com.desiato.puresynth.models.UserDTO;
+import com.desiato.puresynth.services.AuthenticationService;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Optional;
+
+@Component
+public class AuthTokenFilter extends OncePerRequestFilter {
+
+    @Autowired
+    private AuthenticationService authenticationService;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+        String token = request.getHeader("authToken");
+
+        if (isPublicEndpoint(request.getRequestURI())) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        if (token == null) {
```

Comment:
I think null-case should also be handled by `authenticationService.authenticateByToken`.

Reply:
What about this? You haven't responded

Reply:
Yes, I am also handling null-case scenario in ```authenticationService.authenticateByToken```